### PR TITLE
DOC: Improve docs for `jax.numpy`: `deg2rad`, `rad2deg`, `degrees`, `radians`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/ufuncs.py [hunk 1]
```diff
@@ -2016,22 +2016,82 @@ def square(x: ArrayLike, /) -> Array:
...
+  Args:
+    x: scalar or array. Specifies the angle in degrees.
+
+  Returns:
+    An array containing the angles in radians.
+
+  See also:
+    - :func:`jax.numpy.rad2deg` and :func:`jax.numpy.degrees`: Converts the angles
+      from radians to degrees.
+    - :func:`jax.numpy.radians`: Converts the angles from degrees to radians.
```
Comment (jakevdp) [inline jax/_src/numpy/ufuncs.py:2040]:
```suggestion
    - :func:`jax.numpy.radians`: alias of ``deg2rad``.
```

File: jax/_src/numpy/ufuncs.py [hunk 2]
```diff
@@ -2016,22 +2016,82 @@ def square(x: ArrayLike, /) -> Array:
...
+  Args:
+    x: scalar or array. Specifies the angle in radians.
+
+  Returns:
+    An array containing the angles in degrees.
+
+  See also:
+    - :func:`jax.numpy.deg2rad` and :func:`jax.numpy.radians`: Converts the angles
+      from degrees to radians.
+    - :func:`jax.numpy.degrees`: Converts the angles from radians to degrees.
```
Comment (jakevdp) [inline jax/_src/numpy/ufuncs.py:2074]:
```suggestion
    - :func:`jax.numpy.degrees`: alias of ``rad2deg``.
```

File: tests/lax_numpy_test.py [hunk 3]
```diff
@@ -6309,7 +6309,7 @@ def test_lax_numpy_docstrings(self):
     unimplemented = ['fromfile', 'fromiter']
     aliases = ['abs', 'acos', 'acosh', 'asin', 'asinh', 'atan', 'atanh', 'atan2',
                'amax', 'amin', 'around', 'bitwise_right_shift', 'divide', 'pow',
-               'round_']
+               'round_', 'degrees', 'radians']
```
Comment (jakevdp) [inline tests/lax_numpy_test.py:6312]: Let's keep this list alphabetized.

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/ufuncs.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Done

File: tests/lax_numpy_test.py (hunk 3 above)
Comment (rajasekharporeddy) [reply tests/lax_numpy_test.py]: Thanks! Done

**Review by jakevdp — APPROVED** [review]
Thanks! Can you please squash the changes into a single commit?

**Comment by rajasekharporeddy** [comment]
Thanks! Squashed into single commit.

**Review by jakevdp — APPROVED** [review]

**Pull request merged**